src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -247,14 +247,12 @@ def dependencies(
         return (
             entry,
             tuple(
-                dependency_entry
+                entries.get((d.group, d.artifact, d.classifier))
                 for d in entry.dependencies
-                # The dependency might not be present in the entries due to coursier bug:
```

Comment:
I'm not yet confident that the issue has been fixed on the coursier side, just  because the one example we had seems not to occur. Can we point to the coursier PR that truly fixes this?